docs/source/design/huggingface_integration.rst

```diff
@@ -0,0 +1,39 @@
+Integration with HuggingFace
+===================================
+
+This document describes how vLLM integrates with HuggingFace libraries. We will explain step by step what happens under the hood when we run ``vllm serve``.
+
+Let's say we want to serve the popular llama model by running ``vllm serve meta-llama/Llama-3.1-8B``.
+
+- The ``model`` argument is ``meta-llama/Llama-3.1-8B``. vLLM will first try to locate the config file ``config.json`` using this argument. See the `code snippet <https://github.com/vllm-project/vllm/blob/10b67d865d92e376956345becafc249d4c3c0ab7/vllm/transformers_utils/config.py#L75>`__ for the implementation.
```

Comment:
Should we consider making the top level a numbered/ordered list?

Reply:
good point, added in [9d81d85](https://github.com/vllm-project/vllm/pull/10173/commits/9d81d852710006ed462494728b7ed0f6826b5105)